neo/rawio/neuronexusrawio.py

```diff
@@ -0,0 +1,275 @@
+"""
+Neuronexus has their own file format based on their Allego Recording System
+https://www.neuronexus.com/webinar/allego-software-updates/
+
+The format involves 3 files:
+  * The *.xdat.json metadata file
+  * The *_data.xdat binary file of all raw data
+  * The *_timestamps.xdat binary file of the timeestamp data
+  
+Based on sample data is appears that the binary file is always a float32 format
+Other information can be found within the metadata json file
+
+
+The metadata file has a pretty complicated structure as far as I can tell
+a lot of which is dedicated to probe information, which won't be handle at the 
+the Neo level.
+
+It appears that the metadata['status'] provides most of the information necessary
+for generating the initial memory map (global sampling frequency), n_channels,
+n_samples.
+
+metadata['sapiens_base']['biointerface_map'] provides all the channel specific information
+like channel_names, channel_ids, channel_types.
+
+An additional note on channels. It appears that analog channels are called `pri` or
+`ai0` within the metadata where as digital channels are called `din0` or `dout0`.
+In this first implementation it is up to the user to do the appropriate channel slice
+to only get the data they want. 
+
+Zach McKenzie
+
+"""
+
+from __future__ import annotations
+from pathlib import Path
+import json
+
+import numpy as np
+
+from .baserawio import (
+    BaseRawIO,
+    _signal_channel_dtype,
+    _signal_stream_dtype,
+    _spike_channel_dtype,
+    _event_channel_dtype,
+)
+from neo.core import NeoReadWriteError
+
+
+class NeuronexusRawIO(BaseRawIO):
+
+    extensions = ["xdat", "json"]
+    rawmode = "one-file"
+
+    def __init__(self, filename: str | Path = ""):
```

Comment:
```suggestion
    def __init__(self, filename: str | Path):
```
Why add a default value here?

Reply:
It's been like that for ages. Every RawIO has a default. There has been a discussion about switching the default to None, but there is always a default.

Reply:
Any idea why? Seems wrong to me : P

Reply:
No that decision was before my time. Maybe at some point it was useful to initialize the class without an actual file for something in the testing suite? But honestly not sure. I thought it was strange to, but I just live with it now. :)

Reply:
Yes, not that you mentioned it, I think at some point I removed one and it broke down stuff in the part of neo that I don't use : O (the not raw part)